magda-web-client/src/Components/Dataset/Add/DatasetAddCommon.ts

```diff
@@ -1086,6 +1263,70 @@ export async function updateDatasetFromState(
     await updateDataset(datasetRecord, distributionRecords);
 }
 
+type FailedFileInfo = {
+    id?: string;
+    title: string;
+};
+
+/**
+ * Tried to delete any uploaded files that is not associated with any distributions.
+ * Return info of files that are failed to delete
+ *
+ * @export
+ * @param {State} state
+ * @returns {Promise<FailedFileInfo[]>}
+ */
+
+/**
+ * Tried to delete any uploaded files that is not associated with any distributions.
+ * Return info of files that are failed to delete.
+ *
+ * @export
+ * @param {string[]} uploadedFileUrls a list of all files have been uploaded to intnernal magda storage API. Can be retrieved from state.uploadedFileUrls
+ * @param {Distribution[]} distributions all dataset's dsitributions. Can be retrieved from state.distributions
+ * @returns {Promise<FailedFileInfo[]>}
+ */
+export async function cleanUpOrphanFiles(
+    uploadedFileUrls: string[],
+    distributions: Distribution[]
+): Promise<FailedFileInfo[]> {
+    return (
+        await Promise.all(
+            uploadedFileUrls.map(async (fileUrl) => {
+                if (
+                    distributions.find((item) => item.downloadURL === fileUrl)
+                ) {
+                    // --- do nothing if the url is allocated to a distribution
+                    return { isOk: true };
+                }
+
+                let distId, fileName;
+
+                try {
+                    const result = getDistInfoFromDownloadUrl(fileUrl);
```

Comment:
The distId was only used to generate the error message (I think system admin might want to know the distId rather than just filename when a user asks for help).
The file deletion function directly sends the download URL to deletion API. Thus, doesn't use the distId for the deletion.
I removed the usage of `distId` and made `console.error` output `downloadUrl` as well--- I think system admin should be able to capture that with a system like `rollbar`